homeassistant/components/fan/dyson.py

```diff
@@ -238,6 +288,210 @@ def supported_features(self) -> int:
     def device_state_attributes(self) -> dict:
         """Return optional state attributes."""
         return {
-            ATTR_IS_NIGHT_MODE: self.is_night_mode,
-            ATTR_IS_AUTO_MODE: self.is_auto_mode
+            ATTR_NIGHT_MODE: self.night_mode,
+            ATTR_AUTO_MODE: self.auto_mode
             }
+
+
+class DysonPureCoolDevice(FanEntity):
+    """Representation of a Dyson Purecool (TP04/DP04) fan."""
+
+    def __init__(self, hass, device):
+        """Initialize the fan."""
+        _LOGGER.debug("Creating device %s", device.name)
+        self.hass = hass
+        self._device = device
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self.hass.async_add_job(
+            self._device.add_message_listener, self.on_message)
+
+    def on_message(self, message):
+        """Call when new messages received from the fan."""
+        from libpurecool.dyson_pure_state_v2 import DysonPureCoolV2State
+
+        if isinstance(message, DysonPureCoolV2State):
+            _LOGGER.debug("Message received for fan device %s: %s", self.name,
+                          message)
+            self.schedule_update_ha_state()
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the display name of this fan."""
+        return self._device.name
+
+    def turn_on(self, speed: str = None, **kwargs) -> None:
+        """Turn on the fan."""
+        from libpurecool.const import FanSpeed
+
+        _LOGGER.debug("Turn on fan %s", self.name)
+
+        if speed is not None:
+            if speed == FanSpeed.FAN_SPEED_AUTO.value:
```

Comment:
Only fan base component speed values are allowed as user interface. They need to be mapped to the dyson speed values.